recipes/gperf/all/conanfile.py

```diff
@@ -59,6 +59,11 @@ def _build_configure(self):
             autotools.make()
 
     def build(self):
+        if tools.valid_min_cppstd(self, "17"):
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "getline.cc"),
+                                  "register int c = getc (stream);",
+                                  "int c = getc (stream);")
```

Comment:
I think you can create a patch and apply this change unconditionally.
A compiler should be smart enough to put a hot variable in a register.

It looks like upstream already fixed it:
https://git.savannah.gnu.org/gitweb/?p=gperf.git;a=blob;f=lib/getline.cc;h=2d97644d859849c608e6653add0228183434d2fa;hb=HEAD#l58

Reply:
Ok, I have moved my fix to a patch